subprojects/android-test/snackbar-rule/src/main/kotlin/com/avito/android/test/app/second/SnackbarProxyRule.kt

```diff
@@ -0,0 +1,62 @@
+package com.avito.android.test.app.second
+
+import android.annotation.SuppressLint
+import android.widget.TextView
+import com.avito.android.rule.SimpleRule
+import com.avito.android.snackbar.proxy.SnackbarProxy
+import com.avito.android.snackbar.proxy.SnackbarProxyHolder
+import com.avito.android.test.waitFor
+import com.google.android.material.snackbar.Snackbar
+import org.junit.Assert
+import com.google.android.material.R as google_R
+
+class SnackbarProxyRule : SimpleRule() {
+
+    private val proxy = Proxy()
+
+    override fun before() {
+        SnackbarProxyHolder.proxy = proxy
+    }
+
+    fun isShown(text: String) {
+        waitFor {
+            Assert.assertTrue("Snackbar with text=\"$text\" wasn't shown", proxy.snackbarTexts.any { it == text })
+        }
+    }
+
+    fun isShownLast(text: String) {
```

Comment:
Are you sure we need this check? 
It doesn't look like a simple API for me now;

Real world usage is : 'was snackbar with this text(or containtig) shown'? and you probably don't need anything else; besides tap on action

Same goes for `isShownTimes`

Reply:
I will ask our user. I thought it's common behaviour